code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,361 @@
+
+	playsound(src.loc, 'sound/mecha/mechmove03.ogg', 50, TRUE)
+	user.dropItemToGround(src, TRUE)
+	forceMove(gloves)
+
+/obj/item/nemesis_trap
+	name = "disk trap"
+	desc = "A small disk with an energy clamp, designed to be placed on the walls."
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "trap"
+	w_class = WEIGHT_CLASS_SMALL
+
+	var/active = FALSE
+	var/obj/item/clothing/glasses/hud/security/sunglasses/nemesis/glasses
+	var/list/beams = list()
+
+/obj/item/nemesis_trap/afterattack(atom/target, mob/user, proximity, click_parameters)
+	if(!istype(target, /turf/closed))
+		return ..()
+
+	to_chat(user, "<span class='notice'>You begin setting [src] up...</span>")
+	if(!do_after(user, 3 SECONDS, src))
+		return ..()
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(istype(H.get_item_by_slot(ITEM_SLOT_EYES), /obj/item/clothing/glasses/hud/security/sunglasses/nemesis))
+			glasses = H.get_item_by_slot(ITEM_SLOT_EYES)
+
+	active = TRUE
+	icon_state = "trap_deployed"
+	dir = get_dir(target, user)
+	user.dropItemToGround(src, TRUE)
+	update_icon()
+	anchored = TRUE
+	pixel_x = 0
+	pixel_y = 0
+	playsound(get_turf(src), 'sound/magic/blind.ogg', 50, TRUE)
+	deploy_beams()
+
+/obj/item/nemesis_trap/proc/deploy_beams()
+	for(var/obj/item/nemesis_trap/trap in view(3, get_turf(src)))
+		if(trap == src || !trap.active)
+			continue
+
+		icon_state = "trap_deployed_active"
+		update_icon()
+
+		trap.icon_state = "trap_deployed_active"
+		trap.update_icon()
+
+		var/datum/beam/beam = Beam(trap, "nemesis", maxdistance = 5, beam_type = /obj/effect/ebeam/nemesis)
+		beams += beam
+		trap.beams += beam
+
+/obj/item/nemesis_trap/proc/trigger()
+	for(var/datum/beam/beam in beams)
+		qdel(beam)
+	qdel(src)
+
+/obj/effect/ebeam/nemesis
+	name = "energy beam"
+	desc = "An energy beam, connected to a disk trap."
+
+	mouse_opacity = MOUSE_OPACITY_ICON
+
+/obj/effect/ebeam/nemesis/Crossed(atom/movable/AM)
+	. = ..()
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(L.body_position == LYING_DOWN)
+			return
+
+		var/turf/T = get_turf(src)
+		var/obj/item/nemesis_trap/trap_one = owner.origin
+		var/obj/item/nemesis_trap/trap_two = owner.target
+
+		new /obj/effect/temp_visual/nemesis_circle/huge(T)
+
+		to_chat(L, "<span class='userdanger'>You cross [src] and it destabilises, releasing a large energy blast!</span>")
+		playsound(T, 'sound/magic/disable_tech.ogg', 50, TRUE)
+
+		for(var/mob/living/victim in range(1, T))
+			if(victim != L)
+				to_chat(victim, "<span class='userdanger'>You are hit by an energy blast!</span>")
+			victim.apply_damage(35, STAMINA)
+
+		if(trap_one.glasses && ishuman(trap_one.glasses.loc))
+			var/mob/living/carbon/human/H = trap_one.glasses.loc
+			if(H.get_item_by_slot(ITEM_SLOT_EYES) == trap_one.glasses)
+				trap_one.glasses.say("Attention: Disk trap in [get_area(T)] has detonated.")
+
+		trap_one.trigger()
+		trap_two.trigger()
+
+/obj/item/storage/box/nemesis_equipment
+	icon_state = "black_box"
```

Comment:
no name or desc or anything?

Reply:
It's just a common black box filled with stuff from cargo crate, why does it need it?

Reply:
alright i guess